core/trino-main/src/test/java/io/trino/operator/BenchmarkGroupByHash.java

```diff
@@ -415,6 +450,16 @@ private static JoinCompiler getJoinCompiler()
         return new JoinCompiler(TYPE_OPERATORS);
     }
 
+    static {
```

Comment:
> This feels wrong. Why 5 iterations. Feels like this code should be run by JMH as part of the setup. 

It's not an exact science. JMH won't pollute anything if one doesn't write the code. We use similar pattern in other places (e.g `BenchmarkColumnReaders`).

single iteration is not sufficient. JIT is clever enough to ignore initial iterations.

Reply:
Sure it is not. I am not saying that 1 is enough. I was rather saying that 5 is probably also not enough. And if this code is not preprocessed by JMS before execution I am not convinced it is doing anything (or much) useful stuff.

Reply:
> And if this code is not preprocessed by JMS before execution I am not convinced it is doing anything (or much) useful stuff.

It changes numbers though, which is expected, so I guess it works even though it's not nicest

Reply:
JMH has a concept call BulkWarmup, designed specifically for this. It will run warmup iterations for all the benchmarks in a single vm before running the benchmark method

Reply:
Ah, ok. But then I actually need to add benchmarks for all types